examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/DiscoveryExampleFragment.java

```diff
@@ -0,0 +1,390 @@
+
+    ArrayAdapter<CastingPlayer> arrayAdapter =
+        new CastingPlayerCommissionerAdapter(getActivity(), castingPlayerList);
+    final ListView list = getActivity().findViewById(R.id.castingPlayerList);
+    list.setAdapter(arrayAdapter);
+
+    Log.d(TAG, "onViewCreated() creating callbacks");
+
+    this.discoverySuccessCallback =
+        new SuccessCallback<CastingPlayer>() {
+          @Override
+          public void handle(CastingPlayer castingPlayer) {
+            Log.d(
+                TAG,
+                "SuccessCallback handle() CastingPlayer deviceId: " + castingPlayer.getDeviceId());
+            new Handler(Looper.getMainLooper())
+                .post(
+                    () -> {
+                      final Optional<CastingPlayer> playerInList =
+                          castingPlayerList
+                              .stream()
+                              .filter(
+                                  node -> castingPlayer.discoveredCastingPlayerHasSameSource(node))
+                              .findFirst();
+                      if (playerInList.isPresent()) {
+                        Log.d(
+                            TAG,
+                            "Replacing existing CastingPlayer entry "
+                                + playerInList.get().getDeviceId()
+                                + " in castingPlayerList list");
+                        arrayAdapter.remove(playerInList.get());
+                      }
+                      arrayAdapter.add(castingPlayer);
+                    });
+          }
+        };
+
+    Button startDiscoveryButton = getView().findViewById(R.id.startDiscoveryButton);
+    startDiscoveryButton.setOnClickListener(
+        v -> {
+          Log.i(
+              TAG, "onViewCreated() startDiscoveryButton button clicked. Calling startDiscovery()");
+          arrayAdapter.clear();
+          if (!startDiscovery(15)) {
+            Log.e(TAG, "onViewCreated() startDiscovery() call Failed");
+          }
+        });
+
+    Button stopDiscoveryButton = getView().findViewById(R.id.stopDiscoveryButton);
+    stopDiscoveryButton.setOnClickListener(
+        v -> {
+          Log.i(TAG, "onViewCreated() stopDiscoveryButton button clicked. Calling stopDiscovery()");
+          stopDiscovery();
+          Log.i(TAG, "onViewCreated() stopDiscoveryButton button clicked. Canceling future task");
+          scheduledFutureTask.cancel(true);
+        });
+
+    Button clearDiscoveryResultsButton = getView().findViewById(R.id.clearDiscoveryResultsButton);
+    clearDiscoveryResultsButton.setOnClickListener(
+        v -> {
+          Log.i(
+              TAG, "onViewCreated() clearDiscoveryResultsButton button clicked. Clearing results");
+          arrayAdapter.clear();
+        });
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    Log.i(TAG, "onResume() called. Calling startDiscovery()");
+    if (!startDiscovery(15)) {
```

Comment:
Make the 15 a class constant (private static final int DISCOVERY\_RUNTIME\_SEC = 15) in this class.